jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/EcrDockerAction.kt

```diff
@@ -0,0 +1,27 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.ecr.actions
+
+import com.intellij.docker.DockerServerRuntimeInstance
+import com.intellij.openapi.project.DumbAware
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import software.aws.toolkits.jetbrains.core.explorer.actions.SingleExplorerNodeAction
+import software.aws.toolkits.jetbrains.services.ecr.EcrRepositoryNode
+import software.aws.toolkits.jetbrains.services.ecr.EcrUtils
+
+abstract class EcrDockerAction :
+    SingleExplorerNodeAction<EcrRepositoryNode>(),
+    DumbAware
+{
+    abstract val coroutineScope: CoroutineScope
+    protected val dockerServerRuntime by lazy { coroutineScope.dockerServerRuntimeAsync() }
+
+    protected companion object {
+        fun CoroutineScope.dockerServerRuntimeAsync(): Deferred<DockerServerRuntimeInstance> =
+            async(start = CoroutineStart.LAZY) { EcrUtils.getDockerServerRuntimeInstance().runtimeInstance }
```

Comment:
why is this lazy?

Reply:
it creates a default docker connection for the user if it doesn't exist, so i was worried about issues there. maybe it doesn't matter though since it'll always check/create that on the UI thread